### PR TITLE
Allow builds on MSVC

### PR DESCRIPTION
**Pull request by michael-loki**
main bulk is importing cxxopts into `vendor/`

this is 100% a WIP, and I've not tested most of this (because it still doesn't gorram link!)

### CODE REVIEW

File: llarp/win32/abyss.rc [hunk 1]
```diff
@@ -89,7 +89,7 @@ BEGIN
 #ifdef LLARP_RELEASE_MOTTO
             VALUE "ProductVersion", VERSION_STRING(0.4.0, RELEASE_MOTTO, GIT_REV)
 #else
-            VALUE "ProductVersion", VERSION_STRING(0.4.0-dev-, GIT_REV)
```
Comment (despair86) [inline llarp/win32/abyss.rc]: hmmm there should be a way to detect when /lib/cpp is being used as opposed to the K&R era RC preprocessor

File: CMakeLists.txt [hunk 2]
```diff
@@ -320,7 +333,7 @@ if(UNIX)
 elseif(WIN32)
     get_filename_component(LIBTUNTAP_IMPL ${TT_ROOT}/tuntap-windows.c ABSOLUTE)
     get_filename_component(EV_SRC "llarp/ev/ev_win32.cpp" ABSOLUTE)
-    add_definitions(-DWIN32_LEAN_AND_MEAN -DWIN32 -DWINVER=0x500 -D_WIN32_WINNT=0x500)
+    add_definitions(-DWIN32_LEAN_AND_MEAN -DWIN32 -DWINVER=0x0602 -D_WIN32_WINNT=0x0602)
```
Comment (despair86) [inline CMakeLists.txt]: set Windows 8 requirement conditionally, the UNIX tools are slightly dated because they track reactos

**Comment by despair86** [comment]
I love inline libs, we'll keep the new stuff then

File: llarp/win32/abyss.rc (hunk 1 above)
Comment (michael-loki) [reply llarp/win32/abyss.rc]: probably. I'm not sure if this is even needed with MSVC, this is what I started digging into.

File: CMakeLists.txt (hunk 2 above)
Comment (michael-loki) [reply CMakeLists.txt]: 👍 I'm just trying to get to a stage where this builds + tests pass first 😉

File: llarp/nodedb.cpp [hunk 3]
```diff
@@ -173,15 +173,15 @@ llarp_nodedb::Insert(const llarp::RouterContact &rc)
...
 }
 
-ssize_t
+size_t
 llarp_nodedb::Load(const fs::path &path)
 {
   std::error_code ec;
   if(!fs::exists(path, ec))
   {
     return -1;
```
Comment (majestrate) [inline llarp/nodedb.cpp]: this should be set to 0 if the return type is `size_t`

File: llarp/win32/abyss.rc (hunk 1 above)
Comment (michael-loki) [reply llarp/win32/abyss.rc]: reverted.

File: llarp/nodedb.cpp (hunk 3 above)
Comment (michael-loki) [reply llarp/nodedb.cpp]: I'm gonna revert this.

**Comment by michael-loki** [comment]
@despair86 / @majestrate this is ready for merge imo

File: CMakeLists.txt (hunk 2 above)
Comment (despair86) [reply CMakeLists.txt]: I

Comment (michael-loki) [reply CMakeLists.txt]: ah I missed this.

**Comment by despair86** [comment]
ok merging as is, i'll get a patch in shortly

**Pull request merged**